### DISCUSSION TITLE
Render an image accompanied by editorial content

### DISCUSSION THREAD

**Discussion by tomcrane** [digirati-co-uk/iiif-canvas-panel#7] [How Might We]
![image](https://user-images.githubusercontent.com/1443575/128178421-e63a88ad-2992-47a7-8534-372e3633915c.png)

This reflects an editorial requirement. We need to understand what the responsibility of Canvas Panel would be. This might not be a CP concern - "do it yourself in your template environment, and just use CP for the image".

Quire requirement:

> To have a viewer for IIIF images that are annotated (either with text or other images or svg shapes) and where the views of those annotations could be controlled with hyperlinks. For example, for an author to be able to write about some aspect of an artwork image, and for us to be able to add a link to that text that will bring that aspect into view and highlighted with further information in a viewer to the side of the text.

In this image the component is similar to #1 - #4 - an image or a region of an image - but accompanied by some additional markup. This is only a CP concern if we consider the text to be rendered by the component - draw a box round the component and it includes both the image and the text. One scenario in which this might happen is if the thing being rendered is a content state annotation and is accompanied by an HTML label, that the component can render alongside the image.

This would be useful higher-level widget provided by the _framework_ because it can be used over and over again in content management scenarios - Quire, WordPress plugin etc.

Need to be able to style it.

**Comment by tomcrane** [comment]
An extension of this - provide a content state link as part of the rendered output, to open the detail on some other web page with a viewer.

That is, the component can render a `<a href="[template-from-property]?iiif-content=[component-generates-this]>[another-property]</a>`

**Comment by stephenwf** [comment]
What we currently have in Atlas is the following:

* Ability to add markup that tracks a point on the viewer (e.g. a `<div />` with an x, y, w, h on a canvas). 
* Ability to scale with the browser (always scaled at 100% regardless of zoom - e.g. annotating a region)
* Ability to scale with content (scale / text size based on zoom level - e.g. content overlay)

Anything outside is as you say, outside of the realm of a framework or tool.

**Reply by tomcrane** [reply]
Yes I suspect that this could be a helper component that uses the framework (that is still described on the CP site as an example)

The "content-state-selector" tool coming later is a more obvious example of this - utility components alongside the framework.

As a `canvas` web component / tag is also likely to be.

**Comment by tomcrane** [comment]
From the Miro board:

> In one version of this, editorial contains a Content State link; targets a "viewer" that is extremely minimalist, designed to highlight something in a canvas, with the annotation body (e.g., the comment), but not offer any navigation out of that location. It wouldn't even look like a viewer to the reader, it would just look like an accompanying figure - in the run of the text, or to the side. It might render a responsive image tag but not deep zoom, it's already looking at what the author wants you to look at, we don't want you to be distracted by exploration.

> Editorial/User: Text annotated with IIIF links to a viewer or an image set, for example in the text "Cézanne's still life paintings and landscape paintings both", [still life paintings] and [landscape paintings] link to different subsets of a collection. Further text links display annotation layers on an individual image, crop zoom, et cetera.

> Options...   Define the viewport shape/size, as well as the target.   Allow an SVG shape to be clipped on its path (a cutout), highlighted, or nothing (just used to establish a bounding rectangle without rendering it.

**Comment by tomcrane** [comment]
Proposal

This is one of many widgets (that may themselves be web components) that can be developed quickly on top of Canvas Panel.

Some variations, see #1

In this one, https://quire.getty.edu/iiif-snippets/boy-with-straw-hat.json is a content state at a URL. This will have been made by an editor at content-creation time, using #18. Loading it will load a full JSON content state that looks like this:

```json
{
  "type": "Annotation",
  "motivation": ["contentState"],
  "target": {
    "id": "https://iiifmediawiki.herokuapp.com/presentation/canvas/c208117.json#xywh=50,990,2100,1755",
    "type": "Canvas",
    "partOf": [
      {
        "id": "https://iiifmediawiki.herokuapp.com/presentation/File:Baigneurs_a_Asnieres.jpg",
        "type": "Manifest"
      }
    ]
  }
}
```

The content-management template author will then produce code that will output something like this:

```html
<div class="canvas-figure">
    <canvas-panel 
          render="static"
          iiif-content="https://quire.getty.edu/iiif-snippets/
asnieres-detail-boy-with-straw-hat.json"
           />
    <p class="figure-text">
       Fig. 75<br/>
       Georges Seurat (1859-1891)
       <strong>Bathers at Asnières 1884 (detail)</strong><br/>
       <em>Oil on Canvas</em><br/>
       24.1 × 31.1 cm (9 1/2 × 12 1/4 in.)<br/>
       The National Gallery, London<br/>
    </p>
</div>
```

That is, the content-managed data for this widget is the figure text, and a content state pointing to the relevant part of an IIIF resource.

As the template author has the content state handy, they can use it to create a link to another page that would let the user explore the painting in mode detail (but still initialised on the boy-with-straw-hat detail):

```html
<a href="https://getty.edu/iiif-viewer?iiif-content=https://quire.getty.edu/iiif-snippets/
asnieres-detail-boy-with-straw-hat.json">View this painting</a>
```

or, make the canvas panel image the link:

```html
<a href="https://getty.edu/iiif-viewer?iiif-content=https://quire.getty.edu/iiif-snippets/
asnieres-detail-boy-with-straw-hat.json">
    <canvas-panel 
          render="static"
          iiif-content="https://quire.getty.edu/iiif-snippets/
asnieres-detail-boy-with-straw-hat.json"
           />
</a>  ​
```

(Both canvas panel and the anchor tag need accessibility attributes here, omitted for brevity)

An alternative would be where the developer has the region and IIIF information to hand directly, in which case they don't need a stored content state, or they are using the info from the content state directly:

```html
<canvas-panel
   render="static"
   iiif-content="https://iiifmediawiki.herokuapp.com/presentation/canvas/c208117.json"
   partof="https://iiifmediawiki.herokuapp.com/presentation/File:Baigneurs_a_Asnieres.jpg"
   region="50,990,2100,1755">
</canvas-panel>
```

**Comment by tomcrane** [comment]
impl note - the image service in the above example is level 0, sizes only - **no tiles**.

For a static render with a crop, CP could pick an appropriate size and crop client side with css - even when tiles are available, it could still just pick a size, especially when it's a small, locked, illustration detail like this.

**Reply by tomcrane** [reply]
https://codesandbox.io/s/atlas-image-sizes-demo-59exd?file=/src/App.js shows Atlas handling of a sizes-only service. (not a demo of this how-might-we).
